crates/fluvio-benchmark/src/benchmark_config/benchmark_matrix.rs

```diff
@@ -167,3 +240,36 @@ pub fn get_config_from_file(path: &str) -> Vec<BenchmarkMatrix> {
     let file = File::open(path).unwrap();
     vec![serde_yaml::from_reader::<_, BenchmarkMatrix>(file).unwrap()]
 }
+
+#[derive(Debug, Serialize, Deserialize, Copy, Clone, Hash, PartialEq, Eq)]
+pub enum DeliverySemanticStrategy {
+    AtMostOnce,
+    AtLeastOnceFixed,
```

Comment:
Instead of flattening, it would be easier understand by creating At least Once Enum:
```
pub enum DeliverySemanticsStrategy {
    AtMostOnce
    AtLeastOnceStrategy
}

pub enum AtLeastOnceStrategy {
     Fixed,
     Exponential,
     Fibonnaci,
     Custom(RetryPolicy)
}
```

Reply:
Agreed.